### ISSUE TITLE
any help to get this to webgl? threejs

### ISSUE THREAD

**Issue by arpu** [tuxalin/water-shader#2]
is it possible to use this shaders  with webgl 2.0 and threejs ?

**Comment by tuxalin** [comment]
Yes, it should be possible to port it w/o too much work. I can try to setup a threejs demo with it.

**Comment by arpu** [comment]
this would be awesome!

**Comment by arpu** [comment]
@tuxalin  any news on this :> ?

**Comment by tuxalin** [comment]
Sadly I didn't had much free time in the past months, but I do plan to port it one day.

**Comment by arpu** [comment]
testing some Foam water with THREE if this helps https://codesandbox.io/s/hungry-nash-zr8h7?file=/src/index.js

**Comment by tuxalin** [comment]
Thanks! That's actually quite helpful, will use it.

**Comment by nhasdarjian** [comment]
hey @tuxalin, I'm working on a procedural island generator in Three.js and I stumbled across this project while looking to upgrade my water. I have a system set up to load .glsl files, but I am not sure how to go about converting the SPIR-V .spv files to get the plain .glsl.
Is there any chance that you'd be willing to upload the .glsl as plaintext? I'd be glad to share the result with you! Thanks :)

**Comment by tuxalin** [comment]
@nhasdarjian I've started working on this, converting to GLSL and adding threejs support, but due to work I didn't had much free time. Yes, I'll provide some GLSL files.

**Comment by arpu** [comment]
this is awesome! any plans to push it to a branch?

**Comment by tuxalin** [comment]
Sure, will do that. Wanted to get it to work first on threejs
